### PR TITLE
Improve Logging when scanner not found to include platform details

### PR DESCRIPTION
**Pull request by Greg-Hawkridge**

### CODE REVIEW

**Comment by tobyaharris** [comment]
Can the logical tests on `platform` be next to each other?

It was already there but why test `if match & othertests` before setting match to false? Match is only ever set as a bool, if match is already false it doesn't matter if it enters this condition.

File: barcode_dc/code/barcode_scan.py [hunk 1]
```diff
@@ -101,12 +101,13 @@ def find_scanner(self):
 
                 try:
                     if dev.properties['ID_INPUT_KEYBOARD'] == "1":
-                        _, connection_point = dev.properties['ID_PATH'].split('-usb-')
+                        platform, connection_point = dev.properties['ID_PATH'].split('-usb-')
```
Comment (tobyaharris) [inline barcode_dc/code/barcode_scan.py:104]: If the code is quietly doing string operations on the config (eg split), the config file  needs lots of syntax docs.

File: barcode_dc/code/barcode_scan.py [hunk 2]
```diff
@@ -101,12 +101,13 @@ def find_scanner(self):
...
                 try:
                     if dev.properties['ID_INPUT_KEYBOARD'] == "1":
-                        _, connection_point = dev.properties['ID_PATH'].split('-usb-')
+                        platform, connection_point = dev.properties['ID_PATH'].split('-usb-')
                         serial_option_1 = dev.properties['ID_SERIAL']
                         serial_option_2 = f"{dev.properties['ID_VENDOR_ID']}_{dev.properties['ID_MODEL_ID']}"
                         logger.info(
                             f"available: {serial_option_1} or "
-                            f"{serial_option_2} on connection point {connection_point.split(':')}")
+                            f"{serial_option_2} on connection point {connection_point.split(':')} "
```
Comment (tobyaharris) [inline barcode_dc/code/barcode_scan.py:109]: The default config (`connection_point=["0","1.3.1"]`) doesn't include a `:`. How is it meant to split by it? It is unclear what this is meant to do.

File: barcode_dc/code/barcode_scan.py (hunk 1 above)
Comment (Greg-Hawkridge) [reply barcode_dc/code/barcode_scan.py:104]: That's not on the config, that's on the os reported device

File: barcode_dc/code/barcode_scan.py (hunk 2 above)
Comment (Greg-Hawkridge) [reply barcode_dc/code/barcode_scan.py:109]: that's not the connection point from the config file that is being split - it is the string from the os. Splitting the os string puts it into the form that is used in the config file. (OS string would be "0:1.3.1:<a bunch of other things>" which is split to ["0","1.3.1","<a bunch of other things>"] and then compared to find a match.

**Pull request merged**